### PR TITLE
Add a PhoneNumber class to handle formatting and actions on numbers

### PR DESCRIPTION
**Pull request by caitlinrhd**
We value collaboration at Invoca, and focus on peer mentorship as a way for the whole team to learn and thrive. As part of this process, we conduct code reviews during our development cycle. In this phase of the interview, we would like you to review and comment on this Pull Request. 

You can assume that this Pull Request was written by an intern, who is new to professional software development.  Please take about 20-30 minutes to review this Pull Request and comment as if you were reviewing an intern's code.

### CODE REVIEW

File: phone_number.rb [hunk 1]
```diff
@@ -0,0 +1,63 @@
+require_relative "./phone_actions"
+
+class PhoneNumber
+  USERNAME = ENV['ACTIONS_USERNAME']
+  PASSWORD = ENV['ACTIONS_PASSWORD']
+
+  def initialize(number)
+    @number = number
+    @number = format_number
```
Comment (lsubak) [inline phone_number.rb:9]: Instead of setting @number, then setting it again, make format_number a pure function by passing number as a parameter to it.
Also, consider defining a separate formatted number variable. In cases of errors, it can be helpful to have the original input saved.

File: phone_number.rb [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    PhoneActions.new(USERNAME, PASSWORD).text!(@number, message)
+  end
+
+  def call(message)
+    PhoneActions.new(USERNAME, PASSWORD).call!(@number, message)
+  end
+
+  def text_and_call(message)
+    PhoneActions.new(USERNAME, PASSWORD).text!(@number, message)
+    PhoneActions.new(USERNAME, PASSWORD).call!(message, message)
```
Comment (lsubak) [inline phone_number.rb:22]: This should be "(@number, message)", yes?

File: phone_number.rb [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+  def to_i
+    n = @number.gsub!(/[^0-9]/, "")
+    n.to_i
+  end
+
+  private
+
+  def format_number
+    "(#{find_area_code}) #{find_exchange}-#{find_line_number}"
```
Comment (lsubak) [inline phone_number.rb:37]: The find_area_code, find_exchange, and find_line_number functions are all mutating the @number variable when you call !gsub on it. It is generally better to work with a local copy to avoid side effects.

However! In general for these three functions, I would use a full regular expression for the formatting. It can cut down on a lot of the code, make it a bit easier to read, and make it easier to handle errors.
This Stack Overflow question has some good-looking examples: https://stackoverflow.com/questions/1142388/better-way-to-extract-phone-number-and-reformat

File: phone_number_spec.rb [hunk 4]
```diff
@@ -0,0 +1,41 @@
+require_relative "./phone_number"
```
Comment (lsubak) [inline phone_number_spec.rb:1]: Since PhoneActions is specifically used in this file, I would add an explicit reference to it, instead of relying on the reference in PhoneNumber.

File: phone_number_spec.rb [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+RSpec.describe PhoneNumber do
```
Comment (lsubak) [inline phone_number_spec.rb:3]: The main thing I'm seeing that's missing here is "unhappy" paths, and general error handling throughout. It's always good to test that things work, but we also want to be able to fail gracefully. For example, we might want to add tests for if we have an input phone number of "nil" or "5738sda", or "111) 111-1111". 

If we have an invalid phone number, or if the RestClient call returns an error (say, if the internet went down for a moment), we'd want to be able to log those errors and return something informative to the user.
